torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5412,10 +5372,9 @@ def make_arg(shape):
     def prod_zeros(dim_select):
         assert len(dim_select) == 2
         result = make_arg(3 * (S,))
-        with torch.no_grad():
-            result.narrow(dim_select[0], 0, 1).narrow(dim_select[1], 1, 1).zero_()
-            result.narrow(dim_select[0], 2, 1).narrow(dim_select[1], 3, 1).zero_()
-            result.narrow(dim_select[0], 4, 1).narrow(dim_select[1], 3, 1).zero_()
+        result.narrow(dim_select[0], 0, 1).narrow(dim_select[1], 1, 1).zero_()
```

Comment:
The result of `narrow()` on a tensor that requires grad is a tensor that requires grad even under the no grad context?

Reply:
The result of narrow is a new tensor which is a view on a part of a tensor, that then is zeroed out in place. As such, the original tensor object is not modified (although its values are)